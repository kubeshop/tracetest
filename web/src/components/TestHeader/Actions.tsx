import {Dropdown, Menu} from 'antd';
import * as S from './TestHeader.styled';

interface IProps {
  testId: string;
  resultId: string;
}

const Actions = ({testId, resultId}: IProps) => {
  return (
    <span
      data-cy="test-header-actions-button"
      className="ant-dropdown-link"
      onClick={e => e.stopPropagation()}
      style={{textAlign: 'right'}}
    >
      <Dropdown
        overlay={
          <Menu>
            <Menu.Item data-cy="download-junit-button" key="download-junit">
              <a href={`/api/tests/${testId}/run/${resultId}/junit.xml`} download>
                Download JUnit
              </a>
            </Menu.Item>
          </Menu>
        }
        placement="bottomLeft"
        trigger={['click']}
      >
        <S.ActionButton />
      </Dropdown>
    </span>
  );
};

export default Actions;
