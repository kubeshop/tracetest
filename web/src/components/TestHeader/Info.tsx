import {Button, Popover, Typography} from 'antd';
import {InfoCircleOutlined} from '@ant-design/icons';

import Date from 'utils/Date';

interface IProps {
  date: string;
  executionTime: number;
  totalSpans: number;
  traceId: string;
}

const Info = ({date, executionTime, totalSpans, traceId}: IProps) => {
  const content = (
    <>
      <div>
        <Typography.Text strong>Trace ID: </Typography.Text>
        <Typography.Text>{traceId}</Typography.Text>
      </div>
      <div>
        <Typography.Text strong>Trace transaction occurred: </Typography.Text>
        <Typography.Text>{Date.format(date, "yyyy/MM/dd 'at' HH:mm:ss")}</Typography.Text>
      </div>
      <div>
        <Typography.Text strong>Execution time: </Typography.Text>
        <Typography.Text>{executionTime} s</Typography.Text>
      </div>
      <div>
        <Typography.Text strong>Total spans: </Typography.Text>
        <Typography.Text>{totalSpans}</Typography.Text>
      </div>
    </>
  );

  return (
    <Popover placement="right" content={content}>
      <Button
        icon={<InfoCircleOutlined style={{color: 'rgba(3, 24, 73, 0.6)'}} />}
        shape="circle"
        size="small"
        type="text"
      />
    </Popover>
  );
};

export default Info;
