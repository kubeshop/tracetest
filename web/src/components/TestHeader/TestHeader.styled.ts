import {LeftOutlined} from '@ant-design/icons';
import {Typography} from 'antd';
import styled from 'styled-components';

export const BackIcon = styled(LeftOutlined)`
  cursor: pointer;
  font-size: ${({theme}) => theme.size.lg};
`;

export const Container = styled.div<{$isWhite?: boolean}>`
  align-items: center;
  background: ${({$isWhite, theme}) => ($isWhite ? theme.color.white : theme.color.background)};
  border-bottom: ${({theme}) => `1px solid ${theme.color.borderLight}`};
  display: flex;
  justify-content: space-between;
  padding: 12px 0;
  width: 100%;
  margin-bottom: 16px;
`;

export const Section = styled.div`
  align-items: center;
  display: flex;
  gap: 14px;
`;

export const Text = styled(Typography.Text).attrs({
  type: 'secondary',
})`
  && {
    font-size: ${({theme}) => theme.size.sm};
    margin: 0;
  }
`;

export const Title = styled(Typography.Title).attrs({level: 2})`
  && {
    margin: 0;
  }
`;
