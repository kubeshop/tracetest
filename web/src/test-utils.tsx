import {FC, ReactElement} from 'react';
import {render, RenderOptions} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import {ThemeProvider} from 'styled-components';

import {theme} from 'constants/Theme.constants';
import {ReduxWrapperProvider} from 'redux/ReduxWrapperProvider';

const Providers: FC = ({children}) => {
  return (
    <ThemeProvider theme={theme}>
      <ReduxWrapperProvider>
        <MemoryRouter>{children}</MemoryRouter>
      </ReduxWrapperProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, {wrapper: Providers, ...options});

// re-export everything
export * from '@testing-library/react';
// override render method
export {customRender as render};
