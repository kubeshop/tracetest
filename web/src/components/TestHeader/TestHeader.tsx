import ResourceCardActions from 'components/ResourceCard/ResourceCardActions';
import * as S from './TestHeader.styled';
import VariableSetSelector from '../VariableSetSelector/VariableSetSelector';

interface IProps {
  description: string;
  id: string;
  shouldEdit: boolean;
  onBack(): void;
  onEdit(): void;
  onDelete(): void;
  onDuplicate(): void;
  title: string;
  runButton: React.ReactNode;
}

const TestHeader = ({description, id, shouldEdit, onBack, onEdit, onDelete, onDuplicate, title, runButton}: IProps) => (
  <S.Container $isWhite>
    <S.Section>
      <a onClick={onBack} data-cy="test-header-back-button">
        <S.BackIcon />
      </a>
      <div>
        <S.Title data-cy="test-details-name">{title}</S.Title>
        <S.Text>{description}</S.Text>
      </div>
    </S.Section>
    <S.Section>
      <VariableSetSelector />
      {runButton}
      <ResourceCardActions
        id={id}
        onDuplicate={onDuplicate}
        onDelete={onDelete}
        onEdit={onEdit}
        shouldEdit={shouldEdit}
      />
    </S.Section>
  </S.Container>
);

export default TestHeader;
