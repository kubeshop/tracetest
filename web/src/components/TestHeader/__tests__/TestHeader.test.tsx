import {render} from 'test-utils';
import TestMock from '../../../models/__mocks__/Test.mock';
import TestHeader from '../TestHeader';

test('SpanAttributesTable', () => {
  const test = TestMock.model();

  const onBack = jest.fn;
  const onDelete = jest.fn;
  const onEdit = jest.fn;
  const shouldEdit = true;

  const {getByTestId} = render(
    <TestHeader
      description={test.description}
      id={test.id}
      shouldEdit={shouldEdit}
      onBack={onBack}
      onEdit={onEdit}
      onDelete={onDelete}
      title={test.name}
      runButton={<div />}
      onDuplicate={jest.fn()}
    />
  );
  expect(getByTestId('test-details-name')).toBeInTheDocument();
});
