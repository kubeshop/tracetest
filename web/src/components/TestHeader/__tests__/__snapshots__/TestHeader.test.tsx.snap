// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpanAttributesTable 1`] = `
<div>
  <div
    class="sc-dkPtRN giofdF"
  >
    <div
      class="sc-iCfMLu fxGWsx"
    >
      <span
        aria-label="left"
        class="anticon anticon-left sc-gKclnd dGddRj"
        data-cy="test-header-back-button"
        role="img"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          data-icon="left"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
          />
        </svg>
      </span>
      <h4
        class="ant-typography sc-hKwDye gXTdui"
        data-cy="test-details-name"
      >
        
      </h4>
      <span
        class="ant-typography ant-typography-secondary sc-eCImPb bTklbz"
      >
        GET
         - 
        
      </span>
    </div>
    <div
      class="sc-furwcr kLOQYj"
      data-cy="test-run-result-status"
    >
      <span
        class="ant-typography sc-jRQBWg LTTZG"
      >
        Test status:
      </span>
      <div
        class="sc-bdvvtL hICooe"
      >
        <span
          class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
        >
          <span
            class="ant-badge-status-dot ant-badge-status-default"
          />
          <span
            class="ant-badge-status-text"
          >
            Created
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
